File: src/main/java/com/tao/student/Student.java
package com.tao.student;

public class Student {
    String id;
    String name;
    int english;
    int math;
    static int pass=60;//類別層級的定義+static

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public float heightest(){ //最高的成績

       int max= (english > math) ? english:math;
//       if (english > math){
//           max=english;
//       }else{
//           max=math;
//       }
       return max;
    }

    public int getAverage(){
        return (english+math)/2;
    }

    public void print(){

        System.out.print(name+"\t"+english+"\t"+math+"\t"
                           +getAverage());

        if (getAverage() >= pass) {
            System.out.print("\t YOU PASS");
        }else {
            System.out.print("\t you failed");
        }

        char grading='F';
        switch (getAverage()/10){
            case 9:
                grading='A';
                break;
            case 8:
                grading='B';
                break;
            case 7:
                grading='C';
                break;
            case 6:
                grading='D';
                break;
            default:
                grading='F';
        }
      System.out.println("\t grading:"+grading);

    }

}

File: src/main/kotlin/com/tao/student/Student.kt
package com.tao.student

import java.util.*

fun main(args: Array<String>) {

    Student1.pass=50//類別層級

    //userinput()
    val stu = Student1("name", 77,88)
   //stu.print()
    val stu1 = Student1("name1", 55,55)
    stu.print()
    stu1.print()

   // System.out.println("\t"+"成績等級"+stu.grading() )
    System.out.println("最高分數成績："+"\t"+stu.hightest())
}
class Student1(var name:String,var english:Int,var math:Int) {

     companion object{ //設定類別層級的方式
         var pass=60
     }

    fun print() {
        System.out.print(name + "\t" + english + "\t" + math + "\t"
                + getAverage() + "\t" +
                if (getAverage() > pass) "PASS" else "FAILED"  )

        System.out.println("\t成績等級\t"+grading())
    }

 fun grading(): Char{
      var grading=when(getAverage())
      {
    in 90..100 ->'A'
    in 80..89 -> 'B'
    in 70..79 -> 'C'
    in 60..69 -> 'D'
    else -> 'F'
}
    return grading
    }

    fun getAverage():Int{

        return  (english+math)/2
    }

     fun hightest():Int{

         var max=if (english >math)
                    {   System.out.print("english")
                        english}
                        else
                        { print("math")
                            math     }
         return max
    }


private fun userinput() {
    val scanner = Scanner(System.`in`)
    System.out.print("請輸入姓名")
    val name = scanner.next()

    System.out.print("請輸入英文成績")
    val english = scanner.nextInt()

    System.out.print("請輸入數學成績")
    val math = scanner.nextInt()

    val stu = Student1(name, english, math)
    stu.print()
}


}
